CMakeLists.txt

```
@@ -34,9 +34,10 @@ include(GNUInstallDirs)
 include(CheckCCompilerFlag)
 include(CheckCXXCompilerFlag)
 
-set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR}/Bin/${CMAKE_BUILD_TYPE}/)
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR}/Bin/${CMAKE_BUILD_TYPE}/)
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR}/Bin/${CMAKE_BUILD_TYPE}/)
+set(BUILD_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/Bin/" CACHE PATH "Build results output directory")
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${BUILD_OUTPUT_DIRECTORY})
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${BUILD_OUTPUT_DIRECTORY})
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${BUILD_OUTPUT_DIRECTORY})
```

One thing you can do instead then is to use `CMAKE_BINARY_DIR` to make something like
```suggestion
set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/Bin")
set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/Bin")
set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/Bin")
```
Since I'm not sure if adding another variable just for setting where you want the build output files to go to

---

The reason I did this was to be able to keep the old paths for the scripts. Alternatively I could just adapt the CI and documentation to the new path, which is probably a better long-term solution.

---

> adapt the CI and documentation to the new path, which is probably a better long-term solution.

I personally think this would be much better overall since writing to the "source" dir negates and usefulness of out of source builds

---

I absolutely agree, I will adapt the patchset.

---

Is there any difference between `PROJECT_BINARY_DIR` and `CMAKE_BINARY_DIR` btw?

---

I guess the difference would come from if you make SVT-AV1 as a subproject. So `PROJECT_BINARY_DIR` will be set by the most recent `project(name)` cmakelist command while `CMAKE_BINARY_DIR` will by set by the most recent `cmake` command

---

Ok. I do not have much experience with CMake, should we prefer `PROJECT_BINARY_DIR` in order to not put files in the parent project build dir when it is used as subproject or just use `CMAKE_BINARY_DIR`?

---

I guess it would be better to go ahead and just use `PROJECT_BINARY_DIR`, there isn't any harm in choosing it

---

Done the requested change.